### ISSUE TITLE
Stop URL on < or >

### ISSUE THREAD

**Issue by tsl0922** [robinst/autolink-java#7]
code to reproduce:

```java
String input = "wow <p>http://test.com</p> such linked";
LinkExtractor linkExtractor = LinkExtractor.builder().build();
Iterable<LinkSpan> links = linkExtractor.extractLinks(input);
String result = Autolink.renderLinks(input, links, (link, text, sb) -> {
    sb.append("<a href=\"");
    sb.append(text, link.getBeginIndex(), link.getEndIndex());
    sb.append("\">");
    sb.append(text, link.getBeginIndex(), link.getEndIndex());
    sb.append("</a>");
});
System.out.println(result);
```

expect:

```html
wow <p><a href="http://test.com">http://test.com</a></p> such linked
```
actual:

```html
wow <p><a href="http://test.com</p>">http://test.com</p></a> such linked
```

This seems to be a bug, do we support the `skip_tags` feature of rinku?

**Comment by robinst** [comment]
We don't try to parse HTML, this library is about detecting links in plain text.

Having said that, unescaped `<` or `>` is not actually valid in a URL (or IRI) according to [RFC 3987 section 2.2](https://tools.ietf.org/html/rfc3987#section-2.2), so we shouldn't accept it.

I'll make this issue about that.

**Comment by robinst** [comment]
Fixed now, will release soon.

**Comment by robinst** [comment]
Released in 0.5.0 now.
